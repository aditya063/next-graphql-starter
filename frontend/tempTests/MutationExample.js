/* eslint-disable */

import React from 'react';
import { mount } from 'enzyme';
import { ApolloConsumer } from 'react-apollo';
import { MockedProvider } from 'react-apollo/test-utils';
import { load } from '../utils/load';
import * as mock from '../__mocks__/mocks';
import MutationExample, {
  CREATE_USER_MUTATION
} from '../../../components/Examples/MutationExample';
import { USERS_QUERY } from '../../../components/Examples/QueryExample';

describe('MutationExample', () => {
  let mockProps;
  let mockQueries;
  let wrapper;

  beforeEach(() => {
    jest.resetAllMocks();
    mockProps = {};
    mockQueries = [
      {
        request: {
          query: CREATE_USER_MUTATION,
          variables: { name: mock.user.name }
        },
        result: {
          data: {
            createUser: {
              id: mock.user.id,
              name: mock.user.name
            }
          }
        }
      },
      {
        request: { query: USERS_QUERY },
        result: {
          data: {
            users: mock.users
          }
        }
      }
    ];
    wrapper = mount(
      <MockedProvider mocks={mockQueries} addTypename={false}>
        <MutationExample {...mockProps} />
      </MockedProvider>,
      {
        disableLifecycleMethods: true
      }
    );
  });

  it('matches snapshot', () => {
    const mutationExample = wrapper.find('MutationExample');
    expect(mutationExample).toMatchSnapshot();
  });

  it('matches snapshot while mutation loading', async () => {
    wrapper
      .find('input')
      .simulate('change', { target: { value: mock.user.name } });
    wrapper.find('form').simulate('submit');

    expect(wrapper.text()).toContain('Loading...');

    const mutationExample = wrapper.find('MutationExample');
    expect(mutationExample).toMatchSnapshot();
  });

  fit('matches snapshot after mutation completes', async () => {
    wrapper
      .find('input')
      .simulate('change', { target: { value: mock.user.name } });
    wrapper.find('form').simulate('submit');

    await load(wrapper);

    expect(wrapper.text()).not.toContain('Loading...');

    const mutationExample = wrapper.find('MutationExample');
    expect(mutationExample).toMatchSnapshot();
  });

  it('updates state on form submit', async () => {
    const component = wrapper.find('MutationExample').instance();

    expect(component.state.name).toEqual('');

    wrapper
      .find('input')
      .simulate('change', { target: { value: mock.user.name } });
    wrapper.find('form').simulate('submit');

    expect(component.state.name).toEqual(mock.user.name);
  });

  xit('refetches updated data', async () => {
    let apolloClient;

    wrapper = mount(
      <MockedProvider mocks={mockQueries} addTypename={false}>
        <ApolloConsumer>
          {client => {
            apolloClient = client;
            return <MutationExample {...mockProps} />;
          }}
        </ApolloConsumer>
      </MockedProvider>,
      {
        disableLifecycleMethods: true
      }
    );

    const users = await apolloClient.query({ query: USERS_QUERY });

    expect(users.data.users).toEqual(mock.users);
  });

  xit('matches snapshot on error', async () => {
    mockQueries = [
      {
        request: {
          query: CREATE_USER_MUTATION,
          variables: { name: mock.user.name }
        },
        result: {
          data: {
            createUser: {
              id: mock.user.id,
              name: mock.user.name
            }
          }
          // GraphQL error
          // errors: [{ message: 'Error!' }]
        }
        // Network error
        // error: Error('aw shucks')
      },
      {
        request: { query: USERS_QUERY },
        result: {
          data: {
            users: [mock.user]
          }
        }
      }
    ];

    wrapper = mount(
      <MockedProvider mocks={mockQueries} addTypename={false}>
        <MutationExample {...mockProps} />
      </MockedProvider>,
      {
        disableLifecycleMethods: true
      }
    );

    // console.log('debug1: ', wrapper.debug());

    wrapper
      .find('input')
      .simulate('change', { target: { value: mock.user.name } });
    wrapper.find('form').simulate('submit');

    // console.log('debug2: ', wrapper.debug());

    // await new Promise(resolve => setTimeout(resolve, 1000));
    // wrapper.update();

    // console.log('debug3: ', wrapper.debug());

    // expect(wrapper.text()).toContain('Error');

    // const mutationExample = wrapper.find('MutationExample');
    // expect((mutationExample)).toMatchSnapshot();
  });
});
