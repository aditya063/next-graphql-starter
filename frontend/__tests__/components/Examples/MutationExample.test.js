import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { MockedProvider } from 'react-apollo/test-utils';
import MutationExample from '../../../components/Examples/MutationExample';

describe('MutationExample', () => {
  let mockProps;
  let mockQueries;
  let mockMutations;
  let wrapper;

  beforeEach(() => {
    jest.resetAllMocks();
    mockProps = {};
    mockQueries = [];
    mockMutations = [];
    wrapper = shallow(
      <MockedProvider mocks={(mockQueries, mockMutations)}>
        <MutationExample {...mockProps} />
      </MockedProvider>,
      {
        disableLifecycleMethods: true
      }
    );
  });

  it('matches snapshot', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
