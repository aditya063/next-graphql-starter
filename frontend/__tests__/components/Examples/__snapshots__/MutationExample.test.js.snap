// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MutationExample matches snapshot 1`] = `
<ApolloProvider
  client={
    ApolloClient {
      "cache": InMemoryCache {
        "addTypename": true,
        "cacheKeyRoot": CacheKeyNode {
          "children": null,
          "key": null,
        },
        "config": Object {
          "addTypename": true,
          "dataIdFromObject": [Function],
          "fragmentMatcher": HeuristicFragmentMatcher {},
          "resultCaching": true,
        },
        "data": DepTrackingCache {
          "data": Object {},
          "depend": [Function],
        },
        "maybeBroadcastWatch": [Function],
        "optimisticData": DepTrackingCache {
          "data": Object {},
          "depend": [Function],
        },
        "silenceBroadcast": false,
        "storeReader": StoreReader {
          "cacheKeyRoot": CacheKeyNode {
            "children": null,
            "key": null,
          },
          "executeSelectionSet": [Function],
          "executeStoreQuery": [Function],
        },
        "storeWriter": StoreWriter {},
        "typenameDocumentCache": Map {},
        "watches": Set {},
      },
      "clearStoreCallbacks": Array [],
      "clientAwareness": Object {},
      "defaultOptions": Object {},
      "disableNetworkFetches": false,
      "link": ApolloLink {
        "request": [Function],
      },
      "mutate": [Function],
      "query": [Function],
      "queryDeduplication": true,
      "reFetchObservableQueries": [Function],
      "resetStore": [Function],
      "resetStoreCallbacks": Array [],
      "ssrMode": false,
      "store": DataStore {
        "cache": InMemoryCache {
          "addTypename": true,
          "cacheKeyRoot": CacheKeyNode {
            "children": null,
            "key": null,
          },
          "config": Object {
            "addTypename": true,
            "dataIdFromObject": [Function],
            "fragmentMatcher": HeuristicFragmentMatcher {},
            "resultCaching": true,
          },
          "data": DepTrackingCache {
            "data": Object {},
            "depend": [Function],
          },
          "maybeBroadcastWatch": [Function],
          "optimisticData": DepTrackingCache {
            "data": Object {},
            "depend": [Function],
          },
          "silenceBroadcast": false,
          "storeReader": StoreReader {
            "cacheKeyRoot": CacheKeyNode {
              "children": null,
              "key": null,
            },
            "executeSelectionSet": [Function],
            "executeStoreQuery": [Function],
          },
          "storeWriter": StoreWriter {},
          "typenameDocumentCache": Map {},
          "watches": Set {},
        },
      },
      "version": "2.4.13",
      "watchQuery": [Function],
    }
  }
>
  <MutationExample />
</ApolloProvider>
`;
